site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -1161,6 +1161,10 @@ private function updateGradeable(Gradeable $gradeable, $details) {
             }
         }
 
+        if (!$gradeable->hasDueDate() && $gradeable->hasReleaseDate()) {
```

Comment:
Does the front-end allow an instructor to configure their gradeable in this way?

Reply:
The front end will hide the release date option once there is no due date. If both were selected yes at the same time and you only checked the due date to no then this case can exist I believe. I guess that could be handled in the JavaScript instead if you think that is a better option.

Reply:
It's good to have a check like this on the backend (in case they do some POST hack or whatever), but the front-end should also be updated so that they can't have `hasDueDate` be false and `hasReleaseDate` true since it's just going to confuse the instructor when it gets auto-changed on save.